File: Mcma.Api/McmaApiRequest.cs
using System.Collections.Generic;
using Mcma.Core;
using Newtonsoft.Json.Linq;

namespace Mcma.Api
{
    public class McmaApiRequest
    {
        public string Path { get; set; }

        public string HttpMethod { get; set; }
                
        public IDictionary<string, string> Headers { get; set; }
        
        public IDictionary<string, object> PathVariables { get; set; }

        public IDictionary<string, string> QueryStringParameters { get; set; }

        public string Body { get; set; }

        public JToken JsonBody { get; set; }
    }
}

File: nodejs/base/api/lib/http/mcma-api-request.ts
import { McmaApiRequestConfig } from "./mcma-api-request-config";

export class McmaApiRequest extends McmaApiRequestConfig {
    constructor(config: McmaApiRequestConfig) {
        super();

        this.id = config?.id ??  null;
        this.path = config?.path ?? null;
        this.httpMethod = config?.httpMethod ?? null;
        this.headers = config?.headers ?? {};
        this.pathVariables = config?.pathVariables ?? {};
        this.queryStringParameters = config?.queryStringParameters ?? {};
        this.body = config?.body ?? null;
    }
}
